bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/texteditor/FindNextAction.java

```diff
@@ -340,12 +340,10 @@ private int findAndSelect(int offset, String findString, boolean forwardSearch,
 	 * @return the dialog settings to be used
 	 */
 	private IDialogSettings getDialogSettings() {
-		IDialogSettings settings = PlatformUI.getDialogSettingsProvider(FrameworkUtil.getBundle(FindNextAction.class))
+		IDialogSettings settings = PlatformUI
+				.getDialogSettingsProvider(FrameworkUtil.getBundle(FindReplaceAction.class))
 				.getDialogSettings();
-		fDialogSettings= settings.getSection(FindReplaceDialog.class.getName());
-		if (fDialogSettings == null)
-			fDialogSettings= settings.addNewSection(FindReplaceDialog.class.getName());
```

Comment:
This will break the `FindReplaceDialog`, as it still uses the subsection according to its class name. Shouldn't we use a subsection in the dialog settings of the bundle, e.g., using the class name of the `FindReplaceAction` class, like done with the `FindReplaceDialog` before?

Reply:
I have made it so that FindReplaceDialog, FindReplaceOverlay and FindNextAction have the exact same method for getting the Dialog Settings.
I have contemplated adding this as a public method to the FindReplaceAction, however I chose not to in order not to pollute API

Reply:
@Wittmaxi Now you extract different settings sections, as each class (the action, overlay and dialog) retrieve the one according their own class instead of the one for the same class (such as the action):
```java
IDialogSettings dialogSettings = settings.getSection(getClass().getName());
```